code/modules/research/experimentor.dm

```diff
@@ -48,14 +48,28 @@
 	return temp_list
 
 /obj/machinery/rnd/experimentor/proc/valid_items()
-	if (!isnull(valid_items))
-		return valid_items
+	RETURN_TYPE(/list)
 
+	if (isnull(valid_items))
+		generate_valid_items_and_item_reactions()
+
+	return valid_items
+
+/obj/machinery/rnd/experimentor/proc/item_reactions()
+	RETURN_TYPE(/list)
+
+	if (isnull(item_reactions))
+		generate_valid_items_and_item_reactions()
+
+	return item_reactions
+
+/obj/machinery/rnd/experimentor/proc/generate_valid_items_and_item_reactions()
```

Comment:
Could you autodoc the two new procs?

Reply:
Sure but for note auto-docs are not necessary unless implementors use them

Reply:
You're technically right but this whole thing is a wasteland tbh